### ISSUE TITLE
Surprise Duplicate Order Items

### ISSUE THREAD

**Issue by kousu** [savoirfairelinux/santropol-feast#836]
### Expected Behaviour

Creating an order should create one row per *item group*.

### Actual Behaviour

It is possible to create, e.g., multiple Puddings. They are not coalesced into a single row on the route sheets or order page.

### Steps to reproduce

* Create a new order; choose "2" diabetic desserts: ![2018-07-04-163245_1280x800_scrot](https://user-images.githubusercontent.com/987487/42293452-dada405e-7fc9-11e8-89e9-e7e123a28be3.png)

* View the order; there are two
![screenshot_2018-07-04 orders](https://user-images.githubusercontent.com/987487/42293454-e3fa1902-7fc9-11e8-84d3-62db481fdf15.png)

----

Strangely, setting the same value for Diabetic but also adding numbers for the other fields works correctly:

![screenshot_2018-07-04 place orders](https://user-images.githubusercontent.com/987487/42293510-4d147c16-7fca-11e8-80cb-71c2be3b4434.png)

![screenshot_2018-07-04 orders](https://user-images.githubusercontent.com/987487/42293512-4f489ecc-7fca-11e8-80a1-72296353bd1e.png)

**Comment by erozqba** [comment]
Good catch! Could you investigate further into the code? I will try to look to.

**Comment by JeanFrancoisFournier** [comment]
This behavior is probably fine. 

It really has to do with the fact that the first side dish computed is "non-billable" and comes free with the meal. 

Ex. an order of 1 Main, 1 Pudding would show up as 2 items. The pudding being "non-billable".

All other addition of side dishes (even if it's just another unit of the same side dish ordered and flagged as "non-billable" are considered EXTRAS and should be billed. As the value of "billable" can only be a TRUE or FALSE value, all extras will have a separate item entry.

Ex. an order of 1 Main, 2 Pudding would show up as 3 items. The first pudding being "non-billable" and the second being "billable"

Does this explain the behavior?
